lib/heap_sort.rb

```diff
@@ -1,8 +1,24 @@
-
+require_relative "./min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity: O(n * log n)
+# Space Complexity: O(1)
+def heapsort(list)
```

Comment:
Since you're building a whole heap of n nodes, this has space complexity of O(n).   Otherwise well done.